src/controllers/AuthPassportController.js

```diff
@@ -0,0 +1,73 @@
+import status from '../config/status';
+import { User } from '../queries';
+import * as helper from '../helpers';
+
+/**
+ * A class to handle users authentication using social media platform
+ */
+export default class AuthPassportController {
+  /**
+   * @param {object} user
+   * @returns {object} an object containing user information
+   */
+  static async signup(user = {}) {
+    const newUser = Object.keys(user).length
+      ? await User.create(helper.passportSocialMediaUser(user))
+      : {};
+
+    if (!newUser.errors && Object.keys(newUser).length > 0) {
+      return {
+        code: status.CREATED,
+        user: newUser,
+        token: helper.tokenGenerator({ id: newUser.id, role: newUser.role })
+      };
+    }
+
+    return {
+      code: newUser.errors ? status.SERVER_ERROR : status.BAD_REQUEST,
+      error: 'account not created'
```

Comment:
create `const` holders for constant values

Reply:
please, what do you mean by const holders?

Reply:
On my understanding those error messages should be defined as constants then they can be reused any where

Reply:
Ok good, I am working on it, thank you